### DISCUSSION TITLE
Convert Unix Timestamp Wrong

### DISCUSSION THREAD

**Discussion by wangrubinrobin** [grafana/grafana-json-datasource#225] [Q&A] [answered]
Why grafana convert my Unix timetamp (seconds) into a wrong time... 
picture 1 is my original date.
picture 2 is grafana convert my date field into time... but in a wrong way..
Is there anyway i could correct it?

![image](https://user-images.githubusercontent.com/92771424/144736912-d309fec2-ef87-45c0-9778-ce0ddcf2815b.png)

![image](https://user-images.githubusercontent.com/92771424/144736949-02239320-3c74-4081-b04a-817e4cd8ea04.png)

**Comment by marcusolsson** [comment, marked as answer]
Grafana assumes microseconds for time fields and there's currently no support for seconds in the plugin unfortunately.

**Comment by irvinseet** [comment]
Hey @marcusolsson , I tried using unix microseconds (16-digit). However, it is converted wrongly. For example,`1689646901000000` is converted to `55512-10-12 17:23:20`. Any idea why?

![jsonapi](https://github.com/grafana/grafana-json-datasource/assets/61399162/88b3e7aa-8f92-4218-8557-cb1fab79a658)
